### ISSUE TITLE
No re-render triggered when root component returns React.Fragment

### ISSUE THREAD

**Issue by AndyBarron** [gaearon/react-hot-loader#799]
Might be a bit edge-case-y, but the following code won't trigger a re-render (even though the console claims to have hot loaded successfully):

```jsx
import { hot } from 'react-hot-loader';
import React from 'react';

const App = () => (
  <React.Fragment>
    <h1>Example</h1>
    <p>No hot reload worky :-(</p>
  </React.Fragment>
);

export default hot(module)(App);
```

It wouldn't be totally unreasonable if this case just didn't work, but if so, that should be documented somewhere 👍

NB: The example above is simplified from my actual use case, so let me know if you can't get repro, and I can share more details.

**Comment by theKashey** [comment]
We actually don’t have a single test for this case, as long they are the same for react 15 and 16. 
Easy to test. I hope easy to solve.

**Comment by neoziro** [comment]
Fixed in 4.0.0-beta.16.
